equinox/nn/_embedding.py

```diff
@@ -143,52 +135,37 @@ def process_heads(
 
                     return query_heads, key_heads, value_heads
 
-                mha_output = self.mha_attention(
-                    ...
-                    process_heads=process_heads,
-                )
+                x = self.mha_attention(... process_heads=process_heads)
+                ...
         ```
 
     ??? cite
 
         [RoFormer: Enhanced Transformer with Rotary Position Embedding](https://arxiv.org/pdf/2104.09864)
 
         ```bibtex
-            @misc{su2023roformer,
-              title={RoFormer: Enhanced Transformer with Rotary Position Embedding},
-              author={Jianlin Su and Yu Lu and Shengfeng Pan and Ahmed Murtadha and
+        @misc{su2023roformer,
+          title={RoFormer: Enhanced Transformer with Rotary Position Embedding},
+          author={Jianlin Su and Yu Lu and Shengfeng Pan and Ahmed Murtadha and
               Bo Wen and Yunfeng Liu},
-              year={2023},
-              eprint={2104.09864},
-              archivePrefix={arXiv},
-              primaryClass={cs.CL}
-            }
+          year={2023},
+          eprint={arXiv:2104.09864},
+        }
         ```
     """
 
     embedding_size: int = field(static=True)
 
-    def __init__(
-        self,
-        embedding_size: int,
-        *,
-        key: Optional[PRNGKeyArray] = None,
-        **kwargs,
-    ):
-        """**Arguments:**
-
-        - `embedding_size`: Size of the token embeddings. Must be non-negative.
-        - `key`: Not used; provided for compatibility with the rest of the Equinox API.
-            (Keyword only argument.)
-        """
-        if embedding_size < 0:
-            raise ValueError("embedding_size must not be negative.")
-        self.embedding_size = embedding_size
+    def __check_init__(self):
+        if self.embedding_size < 0:
+            raise ValueError("`embedding_size` must not be negative.")
+        if (self.embedding_size % 2) != 0:
+            raise ValueError("`embedding_size` must be even.")
 
     @staticmethod
-    def negate_half(x: Float[Array, "seq_len embedding_size"]):
+    def rotate_half(x: Float[Array, "seq_len embedding_size"]):
```

Comment:
Just as a comment: I will change `seq_len` to `seq_length` to match the other mentions of the sequence length.